src/librustc/hir/check_attr.rs

```diff
@@ -93,70 +93,104 @@ struct CheckAttrVisitor<'tcx> {
 impl CheckAttrVisitor<'tcx> {
     /// Checks any attribute.
     fn check_attributes(&self, item: &hir::Item, target: Target) {
-        if target == Target::Fn || target == Target::Const {
-            self.tcx.codegen_fn_attrs(self.tcx.hir().local_def_id(item.hir_id));
-        } else if let Some(a) = item.attrs.iter().find(|a| a.check_name(sym::target_feature)) {
-            self.tcx.sess.struct_span_err(a.span, "attribute should be applied to a function")
-                .span_label(item.span, "not a function")
-                .emit();
-        }
-
+        let mut is_valid = true;
         for attr in &item.attrs {
-            if attr.check_name(sym::inline) {
+            let current_attr_is_valid = if attr.check_name(sym::inline) {
```

Comment:
```suggestion
            is_valid &= if attr.check_name(sym::inline) {
```

Reply:
Imo the current formulation is cleaner because it is amenable to being expressed as a fold (and avoiding `let mut`).

Reply:
(but `.all` is short-circuiting, which we might not want...)

Reply:
We don't: we want to emit all errors.